### ISSUE TITLE
Rule Change: add an option `caseSensitive` for `no-warning-comments` rule

### ISSUE THREAD

**Issue by zanminkian** [eslint/eslint#19286]
### What rule do you want to change?

no-warning-comments

### What change do you want to make?

Generate fewer warnings

### How do you think the change should be implemented?

A new option

### Example code

```js
// todo: lower case `todo` is not bad
// TODO: upper case is ok

```
I want to make codebase consistently use `TODO` rather than `todo`.

### What does the rule currently do for this code?

This rule currently is case insensitive. Users can't configure it to be case sensitive.

### What will the rule do after it's changed?

```js
/*eslint no-warning-comments: ["error", { "terms": ["TODO", "FIXME"], "caseSensitive": true}]*/
// todo: eslint will not report on this line
// TODO: eslint will report on this line
```

### Participation

- [x] I am willing to submit a pull request to implement this change.

**Comment by Tanujkanti4441** [comment]
Hi @zanminkian, thanks for the issue, but the stylistic rules like `no-warning-comments` are [frozen](https://github.com/eslint/eslint?tab=readme-ov-file#stylistic-rule-updates) and not accepting any changes.
